src/server.c

```diff
@@ -4359,6 +4390,47 @@ void bytesToHuman(char *s, unsigned long long n) {
     }
 }
 
+/* An array of time buckets, each representing a latency range,
+ * between 1 nanosecond and roughly 1 second.
+ * Each bucket covers twice the previous bucket s range.
+ * Empty buckets are not printed.
+ * Everything above 1sec is considered +Inf. */
+sds fillCumulativeDistributionLatencies(sds info, const char* histogram_name, struct hdr_histogram* histogram){
+    info = sdscatprintf(info, "latency_hist_usec_%s:calls=%lld,histogram=[",
+        histogram_name, (long long) histogram->total_count);
```

Comment:
I prefer keeping this together on a single line. When we move it to RESP, I imagine we will still want to group the histogram as a map (or list since it's ordered) under the top level command name. We could still flatten it though, so that is parsed naturally by clients.
```
latency_name:calls=%lld, %d=%d, %d=%d
```

You just have to know the first argument is the calls, and the remaining are the histogram classes.

Reply:
Or perhaps just a simpler format would suffice?
```
latency_name:calls=%lld,histogram=[%d=%d,%d=%d]
```
Then when we move to resp it would look like
```
1) Latency_name
2) 1) calls
    2) %d
    3) histogram
    4) 1) %d
       2) %d
       3) %d
       4) %d
```

Reply:
isn't that what we already have?
well, i guess that's ok.. we'll have to remember going forward that `[]` is the way to add nesting into INFO lines.

Reply:
ohh, i see what Madelyn meant. i.e. the format inside the `[]` is the same one as outside of them.
i.e. `,` separated pairs using `=`. well, i guess that's a better option.

p.s. another option is to completely exclude all of that from the INFO command, and put it in the LATENCY command.
if this data is mainly meant to be processed by humans and / or specific code designed to parse that data, then a new sub-command in LATENCY command is preferred.

If on the other hand we intend for generic monitoring systems to capture that data (like someone feeding all of redis's info fields into Prometheus), then INFO is the right place (e.g. that's one place that the MEMORY STATS command got wrong IMHO).

but considering that generic monitoring systems won't be able to handle that anyway (since it's a compound field), then maybe INFO is the wrong place for it.
@madolson @filipecosta90 @yossigo WDYT?

Reply:
agree with @oranagra about splitting this information into two places:

- INFO will hold the per-command percentile data that is easily parsed and integrated into monitoring tools. The `latency-tracking-info-percentiles` configuration changes the default INFO exposed percentiles.
- LATENCY will hold the histograms and will also enable to quickly compute a set of quantiles on the fly for investigation purposes. 
  - LATENCY HISTOGRAM [commandname] : if no commandname is passed then all then we reply all histograms to the client
  - LATENCY PERCENTILE commandname percentile ... will enable to calculate on the fly any percentile for a specific command
 
 agree?

Reply:
I'm good with that.

Reply:
a bit more reasoning behind the above conclusion (form a talk i had with Filipe).

* the histogram is too verbose for INFO, and may be required only in hands-on investigation
* is it unlikely that a monitoring tool (that's only capable of working with INFO) will handle these histograms anyway (since the structure is nested)
* so moving these histograms to a separate commands let's us output that data in RESP, or maybe also add a human-readable variant (maybe LATENCY HISTOGRAM-DOCTOR :smile: )

about the percentile:
* it is calculated on report-time, that's why the config that controls it was added an `info` tag to it's name
* it is much less verbose, and more useful for a quick glance
* it's not a compound format, so any monitoring tool that's able to handle `info keyspace` and `info commandstats` can extract that data.
* since we can report other percentiles at runtime at will, we can add a command to output that, and the config only controls what's in INFO. 

so two suggestion for changed in Filipe's last post:
* LATENCY HISTOGRAM [command <name>]
* LATENCY HISTOGRAM-REPORT [command <name>] (the human readable version)
* LATENCY PERCENTILE [command <name>] [percentile <pct>] ...